packages/@sanity/portable-text-editor/test/web-server/components/Editor.tsx

```diff
@@ -71,6 +71,7 @@ export const Editor = ({
   const selectionString = useMemo(() => JSON.stringify(selectionValue), [selectionValue])
   const editor = useRef<PortableTextEditor>(null)
   const keyGenFn = useMemo(() => createKeyGenerator(editorId.substring(0, 1)), [editorId])
+  const [readOnly, setReadOnly] = useState(!window.navigator.onLine)
```

Comment:
I'm a fan of using a local state override instead of syncing props with state, e.g. instead of updating based on a new props value and triggering a re-render, I'd suggest changing this state to `isOffline` and use that later on with `readOnly={isOffline || readOnlyFromProps}`, which I think also reads better.